packages/core/src/sync/transport.ts

```diff
@@ -45,7 +60,126 @@ export const cachedTransport = ({
+            method: "eth_call",
+            params: [
+              {
+                data: call.callData,
+                to: call.target,
+              },
+              blockNumber,
+            ],
+          })).map((request) =>
+            toLowerCase(JSON.stringify(orderObject(request))),
+          );
+
+          if (requests.length === 0) {
+            return encodeFunctionResult({
+              abi: multicall3Abi,
+              functionName: "aggregate3",
+              // @ts-expect-error known issue in viem
+              result: [[]],
+            });
+          }
+
+          const cachedResults = await syncStore.getRpcRequestResults({
+            requests,
+            chainId: chain!.id,
+          });
+
+          const multicallResult = cachedResults.every(
+            (result) => result !== undefined,
+          )
+            ? []
+            : await requestQueue
+                .request({
+                  method: "eth_call",
+                  params: [
+                    {
+                      to: params[0]!.to,
+                      data: encodeFunctionData({
+                        abi: multicall3Abi,
+                        functionName: "aggregate3",
+                        args: [
+                          multicallData.args[0]!.filter(
+                            (_, index) => cachedResults[index] === undefined,
+                          ),
+                        ],
+                      }),
+                    },
+                    blockNumber!,
+                  ],
+                })
+                .then((result) =>
+                  decodeFunctionResult({
+                    abi: multicall3Abi,
+                    functionName: "aggregate3",
+                    data: result,
+                  }),
+                );
+
+          // Note: insertRpcRequestResults errors can be ignored and not awaited, since
+          // the response is already fetched.
+          syncStore
+            .insertRpcRequestResults({
+              requests: requests
+                .filter((_, index) => cachedResults[index] === undefined)
+                .map((request, index) => ({
+                  request,
+                  result: multicallResult[index]!,
+                }))
+                // Note: we don't cache request that failed or returned "0x". See more about "0x" below.
+                .filter(
+                  ({ result }) => result?.success && result.returnData !== "0x",
+                )
+                .map(({ request, result }) => ({
+                  request,
+                  blockNumber:
+                    blockNumber === undefined
+                      ? undefined
+                      : blockNumber === "latest"
+                        ? 0
+                        : hexToNumber(blockNumber),
+                  result: JSON.stringify(result.returnData),
+                })),
+              chainId: chain!.id,
+            })
+            .catch(() => {});
```

Comment:
Seems like we should at least log a warning here, it's not expected that `insertRpcRequestResults` will ever throw, right?

Reply:
The `database.wrap` inside `insertRpcRequestResults` will log an error